### ISSUE TITLE
Multiple DHTs using the same KHost?

### ISSUE THREAD

**Issue by wegel** [wasabii/Cogito.Kademlia#2]
Hi,

   Good work on this library, it's very nicely laid out, can't wait to play with it more.

   I need to handle multiple DHTs on a single endpoint; the DHT are differentiated by a unique ID in every message. Do you think support for this could be added to this library, and if so, what would need to be modified/added, generally speaking (if I were to try to add support for this myself)? Thanks!

**Comment by wasabii** [comment]
Yeah. That was the whole point of network id at the protocol layer. Will have to look, but I think I should work now. Or, if not, be very easy to finish.

**Comment by wasabii** [comment]
Ok. Took a look. So, it would be two KHosts. And UdpProtocol needs some work. UdpServer was introduced but never finished. The idea would be to create two hosts. Two protocols. But one UdpServer that is shared between them. Receive sockets should be moved to the server. Server should forward to protocols based on network ID.

**Comment by wegel** [comment]
Makes sense. If you don't plan on working on this within the next 2 weeks or so, I'll probably take a stab at this feature, and do a PR if I can manage to get it working.
